Build example solution in the CI

alexandre-allard opened the pull request:
**Component**: ci, build

**Context**: We want to be able to tests the whole Solution framework in the CI

**Summary**: Building of the example-solution in the CI 

**Acceptance criteria**: an example-solution ISO is well generated and available for download in artifacts. 

---

Closes: #1605

alexandre-allard committed:
build: allow the configuration of the docker socket

Allow to change the socket used to communicate with
docker for skopeo during the build of the example
solution
We need it, because we use a unix socket when building
locally, but a tcp one when in the CI.

Refs: #1605

alexandre-allard committed:
ci: add a step to build example-solution

This step will be used to provide an example
solution artifact to be able to tests the
whole Solution framework.

Refs: #1605

alexandre-allard committed:
ci: enable building of example solution in post-merge

First step to be able to tests the Solution framework

Refs: #1605

bert-e commented:
# Hello alexandre-allard-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1605-build-example-solution-in-ci` into
`development/2.4`
* [w/2.5/improvement/1605-build-example-solution-in-ci](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/1605-build-example-solution-in-ci) will be merged into `development/2.5` 
* [w/2.6/improvement/1605-build-example-solution-in-ci](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/1605-build-example-solution-in-ci) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

gdemonet commented on eve/main.yml:
```diff
@@ -571,6 +572,59 @@ stages:
+            chmod +x operator-sdk-$OPERATOR_SDK_VERSION-x86_64-linux-gnu &&
+            sudo mkdir -p /usr/local/bin/ &&
+            sudo cp operator-sdk-$OPERATOR_SDK_VERSION-x86_64-linux-gnu /usr/local/bin/operator-sdk &&
+            rm -f operator-sdk-$OPERATOR_SDK_VERSION-x86_64-linux-gnu
```
If the deps are really that different, would be worth creating another builder container image IMO

gdemonet replied:
We could also argue that building this ISO is really not "important" and only targetted for a single stage in the post-merge (the stage that checks Solutions, which we'll add in #1856). Hence, just build on the worker VM (no need for the Docker-in-Docker trick for the Pod worker).

alexandre-allard replied:
> If the deps are really that different, would be worth creating another builder container image IMO

Yes, I've created another builder.

gdemonet reviewed (commented):
Some comments on how/when we build this ISO

Ebaneck commented on eve/main.yml:
```diff
@@ -485,6 +485,7 @@ stages:
           name: Trigger build of previous version
           stage_names:
             - buildprev
+            - build-example-solution
```
We could make the stage trigger name generic since we are not building the previous version of the example-solution
e.g `Trigger build for previous version and example-solution stages simultaneously` or something along those lines...

alexandre-allard replied:
yep, done

alexandre-allard committed:
ci: add a pod builder definition for example-solution

Create a docker image with all necessary dependencies
to build example-solution in the CI

Refs: #1605

bert-e commented:
# History mismatch



Merge commit #456f9555b3e044ec356906aee51fbc0474afcf84 on the integration branch
`w/2.5/improvement/1605-build-example-solution-in-ci` is merging a branch which is neither the current
branch `improvement/1605-build-example-solution-in-ci` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/1605-build-example-solution-in-ci` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1605-build-example-solution-in-ci` into
`development/2.4`
* [w/2.5/improvement/1605-build-example-solution-in-ci](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/1605-build-example-solution-in-ci) will be merged into `development/2.5` 
* [w/2.6/improvement/1605-build-example-solution-in-ci](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/1605-build-example-solution-in-ci) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

bert-e commented:
# History mismatch



Merge commit #cc492685e98015c5380dd359dfd0c1dc8a9d3d5a on the integration branch
`w/2.5/improvement/1605-build-example-solution-in-ci` is merging a branch which is neither the current
branch `improvement/1605-build-example-solution-in-ci` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/1605-build-example-solution-in-ci` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

alexandre-allard commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/1605-build-example-solution-in-ci` into
`development/2.4`
* [w/2.5/improvement/1605-build-example-solution-in-ci](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/1605-build-example-solution-in-ci) will be merged into `development/2.5` 
* [w/2.6/improvement/1605-build-example-solution-in-ci](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/1605-build-example-solution-in-ci) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

Ebaneck reviewed (approved):
LGTM

alexandre-allard commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye alexandre-allard-scality.
